src/endpoints/mex/mex.token.charts.service.ts

```diff
@@ -0,0 +1,106 @@
+import { Injectable } from "@nestjs/common";
+import { GraphQlService } from "src/common/graphql/graphql.service";
+import { OriginLogger } from "@multiversx/sdk-nestjs-common";
+import { gql } from 'graphql-request';
+import { MexTokenChart } from "./entities/mex.token.chart";
+import { MexTokenService } from "./mex.token.service";
+import { CacheService } from "@multiversx/sdk-nestjs-cache";
+import { CacheInfo } from "src/utils/cache.info";
+
+@Injectable()
+export class MexTokenChartsService {
+  private readonly logger = new OriginLogger(MexTokenChartsService.name);
+
+  constructor(
+    private readonly graphQlService: GraphQlService,
+    private readonly mexTokenService: MexTokenService,
+    private readonly cachingService: CacheService,
+  ) { }
+
+  async getTokenPricesHourResolution(tokenIdentifier: string): Promise<MexTokenChart[]> {
+    return await this.cachingService.getOrSet(
+      CacheInfo.TokenHourChar(tokenIdentifier).key,
+      async () => await this.getTokenPricesHourResolutionRaw(tokenIdentifier),
+      CacheInfo.TokenHourChar(tokenIdentifier).ttl,
+    );
+  }
+
+  async getTokenPricesHourResolutionRaw(tokenIdentifier: string): Promise<MexTokenChart[]> {
+    const tokenExists = await this.checkTokenExists(tokenIdentifier);
+
+    if (!tokenExists) {
+      this.logger.error(`Token ${tokenIdentifier} does not exist or has no trading pair`);
+      return [];
+    }
+
+    const query = gql`
+      query tokenPricesHourResolution {
+        values24h(
+          series: "${tokenIdentifier}",
+          metric: "priceUSD"
+        ) {
+          timestamp
+          value
+        }
+      }
+    `;
+
+    try {
+      const data = await this.graphQlService.getExchangeServiceData(query);
+      return this.convertToMexTokenChart(data?.values24h) || [];
+    } catch (error) {
+      this.logger.error(`An error occurred while fetching hourly token prices for ${tokenIdentifier}`, error);
+      return [];
+    }
+  }
+
+  async getTokenPricesDayResolution(tokenIdentifier: string, after: string): Promise<MexTokenChart[]> {
+    return await this.cachingService.getOrSet(
+      CacheInfo.TokenDailyChar(tokenIdentifier, after).key,
+      async () => await this.getTokenPricesDayResolutionRaw(tokenIdentifier, after),
+      CacheInfo.TokenDailyChar(tokenIdentifier, after).ttl,
+    );
+  }
+
+  async getTokenPricesDayResolutionRaw(tokenIdentifier: string, after: string): Promise<MexTokenChart[]> {
+    const tokenExists = await this.checkTokenExists(tokenIdentifier);
+
+    if (!tokenExists) {
+      this.logger.error(`Token ${tokenIdentifier} does not exist or has no trading pair`);
```

Comment:
also here, I don't think we should log each user bad request. Also, I see that we return `[]` for this case but also for exceptions. Will the empty array (`[]`) be cached for both situations? Because for the one when a token does not exist, it's good, but when we encounter an exception while fetching data from the external service, we should not cache an empty result because of an internal issue